docs/src/userguide/numpy_pythran.rst

```diff
@@ -0,0 +1,41 @@
+.. highlight:: python
+
+.. _numpy-pythran:
+
+**************************
+Pythran as a Numpy backend
+**************************
+
+Using the flag ``--np-pythran``, it is possible to use the `Pythran`_ numpy
+implementation for numpy related operations. One advantage to use this backend
+is that the Pythran implementation uses C++ expression templates to save memory
+transfers and can benefit from SIMD instructions of modern CPU.
+
+This can lead to really interesting speedup in some cases, going from 2 up to
+16, depending on the targeted CPU architecture and the original algorithm.
+
+Please note that this feature is experimental.
+
+Usage example with distutils
+----------------------------
+
+You first need to install Pythran. See its `documentation
+<https://pythonhosted.org/pythran/MANUAL.html>`_ for more information.
+
+Then, simply add ``np_pythran=True`` to the ``cythonize`` call in the related
+setup.py.
+
+Here is an example:
+
+.. code::
+
+  from distutils.core import setup
+  from Cython.Build import cythonize
+  
+  setup(
+      name = "My hello app",
+      ext_modules = cythonize('hello_pythran.pyx', np_pythran=True)
```

Comment:
I'd be ok with adding a special case in the `DistutilsInfo` class that looks for `np_pythran` in addition to `distutils:` and does all the build setup changes right there. Basically, if you use in-module configuration for distutils (and `np_pythran`), then you need to call `cythonize()`, so `cythonize()` can rely on `DistutilsInfo` doing its job, including the setup for `np_pythran`. If `cythonize()` is not used, then the in-module config for `np_pythran` is simply not available (and can thus be made an error) and instead, the `Extension` instance must be configured manually and properly by the user, including the include directory setup.
Does that solve your problem here?